app/src/actions/tasks.ts

```diff
@@ -31,15 +33,24 @@ export function loadTasks() {
 export function addTask(task: Task, onComplete: Function) {
   return (dispatch) => {
     delete task._id;
-    return Tasks.add(task)
-      .subscribe((res) => {
-        if (res.length === 1) {
-          dispatch({
-            type: TASK_ADDED,
-            payload: res[0]
-          });
-        }
-      }, null, onComplete); }
+
+    const taskAdd$ = Tasks.add(task).publish();
```

Comment:
I'll give that a try later today and let you know how it goes, didn't know about .share vs .publish, but pushing it up into taskservice does seem to be cleaner - and not needing to worry about it in other parts of the code that might be consuming it.

I'll chat over lunch - but thanks.